qcore/nhm.py

```diff
@@ -32,50 +60,129 @@ class NHMFault:
         fault surface trace (lon, lat)
     """
 
-    def __init__(self, entry: List[str]):
+    def __init__(self, **kwargs):
         """Creates an NHMFault instance from the given NHM text.
 
         Parameters
         ----------
-        entry : List of str
-            The rows of an NHM file of one fault.
-            Format:
-                Row 1: FaultName
-                Row 2: TectonicType , FaultType
-                Row 3: LengthMean , LengthSigma (km)
-                Row 4: DipMean , DipSigma (deg)
-                Row 5: DipDir
-                Row 6: Rake (deg)
-                Row 7: RupDepthMean , RupDepthSigma (km)
-                Row 8: RupTopMean, RupTopMin RupTopMax  (km)
-                Row 9: SlipRateMean , SlipRateSigma (mm/yr)
-                Row 10: CouplingCoeff , CouplingCoeffSigma (mm/yr)
-                Row 11: MwMedian , RecurIntMedian  (yr)
-                Row 12: Num Locations on Fault Surface
-                Row 13+: Location Coordinates (Long, Lat)
-        """
-        rows = list(map(str.strip, entry.split("\n")))
+        kwargs of all the required parameters for a nhm fault
 
-        def str2floats(line):
-            return list(map(float, line.split()))
+        It is up to the caller to make sure all the required parameters are available
```

Comment:
Not sure I'm a fan of this, I think the init should take all the variables. Now it's very unclear what a NHM fault represents.

Reply:
I agree, I was being a little cheeky. Do you reckon filling the doc string is enough or should kwargs be unpacked?

Reply:
I think all required fields that make up a NHM fault should be specified directly. Don't have a problem with optionals in kwargs (I think). I know its a bit of effort, but will make future use much easier and clearer imo.